pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -492,7 +503,13 @@ func (r *Reconciler) exists() (bool, error) {
 			// this error type bubbles back up to the machine-controller to allow
 			// us to delete machines that were never properly created due to
 			// invalid zone.
-			return false, machinecontroller.InvalidMachineConfiguration(fmt.Sprintf("%s: Machine does not exist", r.machine.Name))
+
+			// If the machine has a node assigned, we know that it actually exists,
+			// but it also has an invalid configuration.
+			if r.machine.Spec.ProviderID != nil {
+				return true, machinecontroller.InvalidMachineConfiguration(fmt.Sprintf("%s: Zone does not exist", r.providerSpec.Zone))
+			}
```

Comment:
I don't think we want this, it creates a weird semantic around the `exists` bool returned here and I think that makes it more complicated at the function call sites
```suggestion
```